armi/reactor/composites.py

```diff
@@ -3057,9 +3063,37 @@ def _getReactionRates(self, nucName, nDensity=None):
             )
             return {"nG": 0, "nF": 0, "n2n": 0, "nA": 0, "nP": 0}
 
+    def __getReactionRatesChildren(self):
+        """
+        A ``getReactionRates`` helper method, to get child objects to this one.
+
+        Returns
+        -------
+        objects : list
+            list of child objects, in a fashion relevant to ``getReactionRates``.
+        """
+        from armi.reactor.reactors import Reactor
+
+        # If this ArmiObject is an Assembly or higher, get all the Blocks
+        objects = []
+        if isinstance(self, Reactor):
+            objects = self.core.getBlocks()
+        elif hasattr(self, "getBlocks"):
+            objects = self.getBlocks()
+
+        # Try to just get the children.
+        if not len(objects):
+            objects in self.getChildren()
+
+        # Not all composite objects are iterable (i.e. components)
+        if not len(objects):
+            objects = [self]
+
+        return objects
```

Comment:
I believe we can replace all of this logic with this suggestion, which (IMO) looks simpler.
```suggestion
        from armi.reactor.components import Component

        return self.getChildren(
            deep=True, predicate=lambda x: isinstance(x, Component)
        )
```

Reply:
I tried that on my local machine, and all your tests pass, and I believe it is still accomplishing the same thing.

Reply:
Let me know what you think here. After we decide, I can approve the PR.

Reply:
I will try it!

That doesn't cover the situation where there are no children, right?

So, the final `objects = [self]` case is missing?